hs_tools_resource/models.py

```diff
@@ -207,19 +207,29 @@ def has_all_required_elements(self):
     def get_required_missing_elements(self):  # show missing required meta
         missing_required_elements = super(ToolMetaData, self).get_required_missing_elements()
 
-        if not self.url_bases.all().first():
-            missing_required_elements.append('App-launching Url Pattern')
-        elif not self.url_bases.all().first().value:
-            missing_required_elements.append('App-launching Url Pattern')
-
-        if not self.supported_res_types.all().first():
-            missing_required_elements.append('Supported Resource Types')
-        elif not self.supported_res_types.all().first().supported_res_types.count() > 0:
-            missing_required_elements.append('Supported Resource Types')
-
-        if self.supported_sharing_status.all().first():
-            if not self.supported_sharing_status.all().first().sharing_status.count() > 0:
-                missing_required_elements.append('Supported Sharing Status')
+        # At least one of the two metadata must exist: Home Page URL or App-launching URL Pattern
+        if (not self.url_bases.all().first() or not self.url_bases.all().first().value) \
```

Comment:
@pkdash 
The user action to set it to an empty string calls update() func on the backend. I personally think deleting this entry in update() is a bit awkward.  And setting it to an empty string is supposed to run faster than removing it (I think). Also I remember I saw inconsistency issue between UI and backend after deleting one element in update() in another res type as UI still thought the entry exist even though it was already gone. So I was thinking keeping the element with empty string should be able to walk around that issue.

Is there any specific reason/benefit you suggest removing the whole entry when the string is empty?